conf/base.config

```diff
@@ -24,150 +24,148 @@ process {
+        memory = { check_max(64.GB * (2 ** (task.attempt - 1)), 'memory') }
+        time   = { check_max(24.h * (2 ** (task.attempt - 1)), 'time') }
     }
     withName: CENTRIFUGE_CENTRIFUGE {
-        cpus   = { check_max (8     * task.attempt, 'cpus'   ) }
-        memory = { check_max (40.GB * task.attempt, 'memory' ) }
-        time   = { check_max (12.h  * task.attempt, 'time'   ) }
+        cpus   = { check_max(8 * task.attempt, 'cpus') }
+        memory = { check_max(40.GB * task.attempt, 'memory') }
+        time   = { check_max(12.h * task.attempt, 'time') }
     }
     withName: KRAKEN2 {
-        cpus   = { check_max (8     * task.attempt, 'cpus'   ) }
-        memory = { check_max (40.GB * task.attempt, 'memory' ) }
-        time   = { check_max (12.h  * task.attempt, 'time'   ) }
+        cpus   = { check_max(8 * task.attempt, 'cpus') }
+        memory = { check_max(40.GB * task.attempt, 'memory') }
+        time   = { check_max(12.h * task.attempt, 'time') }
     }
     withName: KRONA_KTIMPORTTAXONOMY {
-        cpus   = { check_max (8     * task.attempt, 'cpus'   ) }
-        memory = { check_max (20.GB * task.attempt, 'memory' ) }
-        time   = { check_max (12.h  * task.attempt, 'time'   ) }
+        cpus   = { check_max(8 * task.attempt, 'cpus') }
+        memory = { check_max(20.GB * task.attempt, 'memory') }
+        time   = { check_max(12.h * task.attempt, 'time') }
     }
     withName: CAT_DB_GENERATE {
-        memory = { check_max (200.GB * task.attempt, 'memory' ) }
-        time   = { check_max (16.h   * task.attempt, 'time'   ) }
+        memory = { check_max(200.GB * task.attempt, 'memory') }
+        time   = { check_max(16.h * task.attempt, 'time') }
     }
     withName: CAT {
-        cpus   = { check_max (8     * task.attempt, 'cpus'   ) }
-        memory = { check_max (40.GB * task.attempt, 'memory' ) }
-        time   = { check_max (12.h  * task.attempt, 'time'   ) }
+        cpus   = { check_max(8 * task.attempt, 'cpus') }
+        memory = { check_max(40.GB * task.attempt, 'memory') }
+        time   = { check_max(12.h * task.attempt, 'time') }
     }
     withName: GTDBTK_CLASSIFYWF {
-        cpus   = { check_max (10     * task.attempt, 'cpus'   ) }
-        memory = { check_max (128.GB * task.attempt, 'memory' ) }
-        time   = { check_max (12.h   * task.attempt, 'time'   ) }
+        cpus   = { check_max(10 * task.attempt, 'cpus') }
+        memory = { check_max(128.GB * task.attempt, 'memory') }
+        time   = { check_max(12.h * task.attempt, 'time') }
     }
     //MEGAHIT returns exit code 250 when running out of memory
     withName: MEGAHIT {
-        cpus          = { check_megahit_cpus (8, task.attempt  ) }
-        memory        = { check_max (40.GB * task.attempt, 'memory' ) }
-        time          = { check_max (16.h  * task.attempt, 'time'   ) }
+        cpus          = { params.megahit_fix_cpu_1 ? 1 : check_max(8 * task.attempt, 'cpus') }
```

Comment:
Does that work? I remember vaguely that it didn't work to access `params` here, that's why I think I shifted the corresponding functionality to `nextflow.config` ... but maybe that changed?

Reply:
It should do if it's in a closure `{ }`. That's why e.g. `task.attempt` works.

Reply:
Did you test?

Reply:
Hrm wait, this can't work as `check_max` should be gone...